mailpoet/lib/WooCommerce/CouponPreProcessor.php

```diff
@@ -65,6 +66,8 @@ private function addOrUpdateCoupon(array $couponBlock, NewsletterEntity $newslet
     }
     $coupon->set_discount_type($couponBlock['discountType']);
     $coupon->set_amount($couponBlock['amount']);
+    $expiration = (new DateTime())->getCurrentDateTime()->modify("+{$couponBlock['expiryDay']} day")->getTimestamp();
+    $coupon->set_date_expires($expiration);
```

Comment:
Just want to double-check check I understand correctly. This updates the expiration each time when the plugin pre-process a newsletter. So for the welcome email, it prolongs expiration every time when the newsletter is sent. Correct? I don't think it is a big issue. Just want to make sure we prolong the expiration so that the new subscribers have a valid coupon. Maybe an integration test would be nice for that.

Reply:
yes, so basically the actual expiration is the one of the last send out, adding a test makes sense.

Reply:
Adding test is a good idea. I think it would be nice to cover whole class with an integration test. It seems that the class handles all the logic about coupon creation. I assume we will be adding more in the follow-up tasks.